Random tidying up

glesica opened the pull request:
@sarahwalling I did some tidying to start moving some of the non-code stuff along as you finish the code bits. Take a look and feel free to either merge or not. Let me know if you'd like me to change anything I've done.

glesica committed:
Remove object files and binary

glesica committed:
Ignore some common nonsense

glesica committed:
More docs and naively fix AVX calls

glesica commented on NINJA/DistanceCalculatorAVX.cpp:
```diff
@@ -1097,13 +1097,13 @@ void DistanceCalculator::convertAllDNA(){
+	this->TRANSITIONS_MASK = _mm256_set_epi8(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0);
 
-	this->TRANSVERSIONS_MASK = _mm256_set_epi8(2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0);
+	this->TRANSVERSIONS_MASK = _mm256_set_epi8(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0);
```
These probably aren't correct, as we discussed, but they let it build and run so I thought it was worthwhile for now.

sarahsummerfield replied:
```suggestion
	this->TRANSVERSIONS_MASK = _mm256_set_epi8(0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0, 0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0);
```

sarahwalling replied:
```suggestion
	this->TRANSVERSIONS_MASK = _mm256_set_epi8(0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0, 0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0);
```

sarahsummerfield commented on NINJA/DistanceCalculatorAVX.cpp:
```diff
@@ -1097,13 +1097,13 @@ void DistanceCalculator::convertAllDNA(){
 
 	//TODO: these are not doubled yet. Not sure they need to be
-	this->GAPS_COUNT_MASK = _mm256_set_epi8(0, 1, 1, 2, 1, 2, 2, 3, 1, 2, 2, 3, 2, 3, 3, 4);
+	this->GAPS_COUNT_MASK = _mm256_set_epi8(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 2, 1, 2, 2, 3, 1, 2, 2, 3, 2, 3, 3, 4);
```
```suggestion
	this->GAPS_COUNT_MASK = _mm256_set_epi8(0, 1, 1, 2, 1, 2, 2, 3, 1, 2, 2, 3, 2, 3, 3, 4, 0, 1, 1, 2, 1, 2, 2, 3, 1, 2, 2, 3, 2, 3, 3, 4);
```

sarahsummerfield commented on NINJA/DistanceCalculatorAVX.cpp:
```diff
@@ -1097,13 +1097,13 @@ void DistanceCalculator::convertAllDNA(){
+	this->GAPS_COUNT_MASK = _mm256_set_epi8(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 2, 1, 2, 2, 3, 1, 2, 2, 3, 2, 3, 3, 4);
 
-	this->DECOMPRESSED_GAPS = _mm256_set_epi8(255, 252, 243, 240, 207, 204, 195, 192, 63, 60, 51, 48, 15, 12, 3, 0);
+	this->DECOMPRESSED_GAPS = _mm256_set_epi8(255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 252, 243, 240, 207, 204, 195, 192, 63, 60, 51, 48, 15, 12, 3, 0);
```
```suggestion
	this->DECOMPRESSED_GAPS = _mm256_set_epi8(255, 252, 243, 240, 207, 204, 195, 192, 63, 60, 51, 48, 15, 12, 3, 0, 255, 252, 243, 240, 207, 204, 195, 192, 63, 60, 51, 48, 15, 12, 3, 0);
```

sarahsummerfield commented on NINJA/DistanceCalculatorAVX.cpp:
```diff
@@ -1097,13 +1097,13 @@ void DistanceCalculator::convertAllDNA(){
+	this->DECOMPRESSED_GAPS = _mm256_set_epi8(255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 252, 243, 240, 207, 204, 195, 192, 63, 60, 51, 48, 15, 12, 3, 0);
 
-	this->TRANSITIONS_MASK = _mm256_set_epi8(0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0);
+	this->TRANSITIONS_MASK = _mm256_set_epi8(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0);
```
```suggestion
	this->TRANSITIONS_MASK = _mm256_set_epi8(0, 0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0);
```

sarahsummerfield committed:
Changing from 0 padding to doubling

Removed the padding of zeros and doubled the original 16 values to total 32

glesica commented:
Thanks for pushing those changes. We can merge it whenever you're ready.

glesica committed:
Simplify makefile

glesica committed:
Merge branch 'avx' into spruce-up-makefile

glesica committed:
Bad at spelling

glesica committed:
Use the correct directory

glesica committed:
Add build to CI

sarahsummerfield commented on NINJA/DistanceCalculatorAVX.cpp:
```diff
@@ -1097,13 +1097,13 @@ void DistanceCalculator::convertAllDNA(){
+	this->DECOMPRESSED_GAPS = _mm256_set_epi8(255, 252, 243, 240, 207, 204, 195, 192, 63, 60, 51, 48, 15, 12, 3, 0, 255, 252, 243, 240, 207, 204, 195, 192, 63, 60, 51, 48, 15, 12, 3, 0);
 
-	this->TRANSITIONS_MASK = _mm256_set_epi8(0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0);
+	this->TRANSITIONS_MASK = _mm256_set_epi8(0, 0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0);
```
```suggestion
	this->TRANSITIONS_MASK = _mm256_set_epi8(0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0);
```

sarahsummerfield commented on NINJA/DistanceCalculatorAVX.cpp:
```diff
@@ -1097,13 +1097,13 @@ void DistanceCalculator::convertAllDNA(){
+	this->TRANSITIONS_MASK = _mm256_set_epi8(0, 0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0);
 
-	this->TRANSVERSIONS_MASK = _mm256_set_epi8(2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0);
+	this->TRANSVERSIONS_MASK = _mm256_set_epi8(0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0, 0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0);
```
```suggestion
	this->TRANSVERSIONS_MASK = __mm256_set_epi8(2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0);
```

sarahsummerfield committed:
Fixing zero-padding removal

A couple of extra zeros got left in - replacing these with properly doubled vectors with no extra 0s in them!

sarahsummerfield commented on NINJA/DistanceCalculatorAVX.cpp:
```diff
@@ -1097,13 +1097,13 @@ void DistanceCalculator::convertAllDNA(){
+	this->TRANSITIONS_MASK = _mm256_set_epi8(0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 2, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0);
 
-	this->TRANSVERSIONS_MASK = _mm256_set_epi8(2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0);
+	this->TRANSVERSIONS_MASK = __mm256_set_epi8(2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0);
```
```suggestion
	this->TRANSVERSIONS_MASK = _mm256_set_epi8(2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0, 2, 1, 2, 1, 1, 0, 1, 0);
```

sarahsummerfield committed:
Fixing transversion mask initalization

Removing extra underscore

glesica commented:
Ah, good, you fixed it, I must have had a bad merge or something.

glesica committed:
Go back to enumerating files
